components/User.tsx

```diff
@@ -2,29 +2,69 @@ import React from 'react'
 import { View, Text, Image, StyleSheet, TouchableOpacity } from 'react-native'
 import Badges from './Badges'
 import { connect } from 'react-redux'
+import { postHangoutRequest } from '../src/actions/userActions'
 
-const User = props => {
-  return (
-    <TouchableOpacity
-      style={styles.user}
-      onPress={() => props.showProfile(props.user)}
-    >
-      {props.user.profile_photo !== null && (
-        <Image
-          source={{ uri: props.user.profile_photo }}
-          style={styles.user__image}
-        />
-      )}
-      <View style={styles.user__textcontainer}>
-        <Text style={styles.user__name}>{props.user.first_name}</Text>
-        <Text style={styles.user__text}>Level {props.user.lvl}</Text>
-        <Text style={styles.user__text}>XP: {props.user.exp}</Text>
-      </View>
-      <View style={styles.user__badges}>
-        <Badges />
+interface UserLimited {
+  id: string
+  email: string
+  first_name: string
+  profile_photo: string
+  exp: number
+  lvl: number
+  interests: [string]
+}
+interface Props {
+  user: UserLimited
+  currentUserEmail: string
+  sendRequest: Function
+}
+
+class User extends React.Component<Props> {
+  handlePress = async () => {
+    const resStatus = await postHangoutRequest(
+      this.props.currentUserEmail,
+      this.props.user.email
+    )
+    if (resStatus === 200) {
+      const toUser = {
+        first_name: this.props.user.first_name,
+        email: this.props.user.email,
+        profile_photo: this.props.user.profile_photo
+      }
+      this.props.sendRequest(toUser)
+    }
+  }
+  render() {
+    return (
+      // <TouchableOpacity
+      //   style={styles.user}
+      //   // onPress={() => props.showProfile(props.user)}
+      // >
+      <View style={styles.user}>
+        {this.props.user.profile_photo !== null && (
+          <Image
+            source={{ uri: this.props.user.profile_photo }}
+            style={styles.user__image}
+          />
+        )}
+        <View style={styles.user__textcontainer}>
+          <Text style={styles.user__name}>{this.props.user.first_name}</Text>
+          <Text style={styles.user__text}>Level {this.props.user.lvl}</Text>
+          <Text style={styles.user__text}>XP: {this.props.user.exp}</Text>
+        </View>
+        <View style={styles.column}>
+          <View style={styles.user__badges}>
+            <Badges />
+          </View>
+
```

Comment:
Temporary fix for user public profile animation bug: user listing has a button to request hangout